snuba/datasets/replays_processor.py

```diff
@@ -0,0 +1,177 @@
+import logging
+import uuid
+from datetime import datetime
+from typing import Any, Mapping, MutableMapping, Optional, Tuple
+
+from snuba import environment
+from snuba.consumers.types import KafkaMessageMetadata
+from snuba.datasets.events_format import (
+    EventTooOld,
+    enforce_retention,
+    extract_extra_tags,
+    extract_user,
+)
+from snuba.processor import (
+    InsertBatch,
+    MessageProcessor,
+    ProcessedMessage,
+    _as_dict_safe,
+    _ensure_valid_date,
+    _ensure_valid_ip,
+    _unicodify,
+)
+from snuba.utils.metrics.wrapper import MetricsWrapper
+
+logger = logging.getLogger(__name__)
+
+metrics = MetricsWrapper(environment.metrics, "replays.processor")
+from sentry_sdk import capture_exception
+
+EventDict = Mapping[Any, Any]
+RetentionDays = int
+
+
+class ReplaysProcessor(MessageProcessor):
+    PROMOTED_TAGS = {
+        "environment",
+        "sentry:release",
+        "sentry:user",
+        "sentry:dist",
+    }
+
+    def __extract_timestamp(self, field: int) -> datetime:
+        timestamp = _ensure_valid_date(datetime.utcfromtimestamp(field))
+        if timestamp is None:
+            timestamp = datetime.utcnow()
+        return timestamp
+
+    def _structure_and_validate_message(
+        self, message: Mapping[Any, Any]
+    ) -> Optional[Tuple[EventDict, RetentionDays]]:
+
+        event = message
+        data = event["data"]
+        event_type = data.get("type")
+        if event_type != "replay_event":
+            return None
```

Comment:
This seems odd. Why would you expect to have anything other than replays on the Kafka topic that you are consuming ?